src/Cli/dotnet/ToolPackage/ToolPackageDownloader.cs

```diff
@@ -0,0 +1,311 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.CommandLine;
+using System.Reflection;
+using Microsoft.DotNet.Cli.NuGetPackageDownloader;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.ToolPackage;
+using Microsoft.DotNet.Tools;
+using Microsoft.Extensions.EnvironmentAbstractions;
+using NuGet.Client;
+using NuGet.Common;
+using NuGet.ContentModel;
+using NuGet.Frameworks;
+using NuGet.LibraryModel;
+using NuGet.Packaging;
+using NuGet.Packaging.Core;
+using NuGet.ProjectModel;
+using NuGet.Repositories;
+using NuGet.RuntimeModel;
+using NuGet.Versioning;
+using System;
+
+
+namespace Microsoft.DotNet.Cli.ToolPackage
+{
+    internal class ToolPackageDownloader : IToolPackageDownloader
+    {
+        private INuGetPackageDownloader _nugetPackageDownloader;
+        private readonly IToolPackageStore _toolPackageStore;
+
+        // The directory that the tool will be downloaded to
+        protected DirectoryPath _toolDownloadDir;
+
+        // The directory that the tool package is returned 
+        protected DirectoryPath _toolReturnPackageDirectory;
+
+        // The directory that the tool asset file is returned
+        protected DirectoryPath _toolReturnJsonParentDirectory;
+
+        // The directory that global tools first downloaded
+        // example: C:\Users\username\.dotnet\tools\.store\.stage\tempFolder
+        protected readonly DirectoryPath _globalToolStageDir;
+
+        // The directory that local tools first downloaded
+        // example: C:\Users\username\.nuget\package
+        protected readonly DirectoryPath _localToolDownloadDir;
+
+        // The directory that local tools' asset files located
+        // example: C:\Users\username\AppData\Local\Temp\tempFolder
+        protected readonly DirectoryPath _localToolAssetDir;
+
+        public static readonly CliOption<VerbosityOptions> VerbosityOption = CommonOptions.VerbosityOption;
+
+        protected readonly string _runtimeJsonPath;
+
+        public ToolPackageDownloader(
+            IToolPackageStore store,
+            string runtimeJsonPathTest = null
+            // string localToolPath = null
```

Comment:
Delete commented out code, and I'd suggest renaming `runtimeJsonPathTest` to something like `runtimeJsonPathForTests` or `runtimeJsonPath`.